### ISSUE TITLE
Messages of the thread won't load

### ISSUE THREAD

**Issue by detouched** [nylas/N1#3059]
Messages from one of the threads in my inbox doesn't show up when I switch to that thread. The spinners on the messages just spin infinitely.

##### What operating system are you using?
Mac OS X 10.12.1. Oh wait, I should write macOS now 🤔 

##### What version of N1 are you using?
Version 0.4.204-c4a286f (0.4.204-c4a286f)

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
No, but some of the default plugins are enabled.

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
I don't think it is related to the email provider.

##### Is the issue reproducible with a particular attachment, message, signature, etc?
Reproducible with a particular message thread.

##### Description
When I click on a particular thread in the inbox, the messages on the right appear but their content never comes up, i.e. the spinner keeps going. Here's an error from the log that appears each time I click on the message thread:

```
{"host":"dpenkin","timestamp":"2016-11-16T00:19:08.930Z","payload":"Error: Failed to execute 'btoa' on 'Window': The string to be encoded contains characters outside of the Latin1 range.\n    at Error (native)\n    at /Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:188:114\n    at RegExp.[Symbol.replace] (native)\n    at String.replace (native)\n    at MessageItemBody._mergeBodyWithFiles (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:187:23)\n    at MessageItemBody._mergeBodyWithFiles (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:3:59)\n    at MessageItemBody._renderBody (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:106:27)\n    at MessageItemBody._renderBody (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:3:59)\n    at MessageItemBody.render (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:99:16)\n    at MessageItemBody.render (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:3:59)\n    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:796:32)\n    at ReactCompositeComponentWrapper._renderValidatedComponent (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:819:34)\n    at ReactCompositeComponentWrapper._updateRenderedComponent (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:743:36)\n    at ReactCompositeComponentWrapper._performComponentUpdate (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:721:10)\n    at ReactCompositeComponentWrapper.updateComponent (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:642:12)\n    at ReactCompositeComponentWrapper.performUpdateIfNecessary (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:558:12)\n    at Object.performUpdateIfNecessary (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactReconciler.js:158:22)\n    at runBatchedUpdates (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdates.js:151:21)\n    at ReactReconcileTransaction.perform (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/Transaction.js:138:20)\n    at ReactUpdatesFlushTransaction.perform (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/Transaction.js:138:20)\n    at Object.flushBatchedUpdates (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdates.js:173:19)\n    at ReactDefaultBatchingStrategyTransaction.closeAll (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/Transaction.js:204:25)\n    at ReactDefaultBatchingStrategyTransaction.perform (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/Transaction.js:151:16)\n    at Object.batchedUpdates (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactDefaultBatchingStrategy.js:63:19)\n    at Object.enqueueUpdate (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdates.js:201:22)\n    at enqueueUpdate (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdateQueue.js:25:16)\n    at Object.enqueueSetState (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdateQueue.js:210:5)\n    at MessageItemBody.ReactComponent.setState (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactComponent.js:64:16)\n    at /Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:53:24\n    at retrieve.then.output (/Applications/Nylas N1.app/Contents/Resources/app.asar/src/flux/stores/message-body-processor.js:106:9)\n    at tryCatcher (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/util.js:26:23)\n    at Promise._settlePromiseFromHandler (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/promise.js:510:31)\n    at Promise._settlePromiseAt (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/promise.js:584:18)\n    at Promise._settlePromises (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/promise.js:700:14)\n    at Async._drainQueue (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/async.js:123:16)\n    at Async._drainQueues (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/async.js:133:10)\n    at Immediate.Async.drainQueues (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/async.js:15:14)\n    at tryOnImmediate (timers.js:554:5)\n    at processImmediate [as _immediateCallback] (timers.js:533:5)"}
{"host":"dpenkin","timestamp":"2016-11-16T00:19:08.930Z","payload":{"promise":{"isFulfilled":false,"isRejected":true,"rejectionReason":{"stack":"Error: Failed to execute 'btoa' on 'Window': The string to be encoded contains characters outside of the Latin1 range.\n    at Error (native)\n    at /Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:188:114\n    at RegExp.[Symbol.replace] (native)\n    at String.replace (native)\n    at MessageItemBody._mergeBodyWithFiles (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:187:23)\n    at MessageItemBody._mergeBodyWithFiles (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:3:59)\n    at MessageItemBody._renderBody (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:106:27)\n    at MessageItemBody._renderBody (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:3:59)\n    at MessageItemBody.render (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:99:16)\n    at MessageItemBody.render (/Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:3:59)\n    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:796:32)\n    at ReactCompositeComponentWrapper._renderValidatedComponent (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:819:34)\n    at ReactCompositeComponentWrapper._updateRenderedComponent (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:743:36)\n    at ReactCompositeComponentWrapper._performComponentUpdate (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:721:10)\n    at ReactCompositeComponentWrapper.updateComponent (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:642:12)\n    at ReactCompositeComponentWrapper.performUpdateIfNecessary (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactCompositeComponent.js:558:12)\n    at Object.performUpdateIfNecessary (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactReconciler.js:158:22)\n    at runBatchedUpdates (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdates.js:151:21)\n    at ReactReconcileTransaction.perform (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/Transaction.js:138:20)\n    at ReactUpdatesFlushTransaction.perform (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/Transaction.js:138:20)\n    at Object.flushBatchedUpdates (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdates.js:173:19)\n    at ReactDefaultBatchingStrategyTransaction.closeAll (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/Transaction.js:204:25)\n    at ReactDefaultBatchingStrategyTransaction.perform (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/Transaction.js:151:16)\n    at Object.batchedUpdates (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactDefaultBatchingStrategy.js:63:19)\n    at Object.enqueueUpdate (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdates.js:201:22)\n    at enqueueUpdate (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdateQueue.js:25:16)\n    at Object.enqueueSetState (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactUpdateQueue.js:210:5)\n    at MessageItemBody.ReactComponent.setState (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/react/lib/ReactComponent.js:64:16)\n    at /Applications/Nylas N1.app/Contents/Resources/app.asar/internal_packages/message-list/lib/message-item-body.js:53:24\n    at retrieve.then.output (/Applications/Nylas N1.app/Contents/Resources/app.asar/src/flux/stores/message-body-processor.js:106:9)\n    at tryCatcher (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/util.js:26:23)\n    at Promise._settlePromiseFromHandler (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/promise.js:510:31)\n    at Promise._settlePromiseAt (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/promise.js:584:18)\n    at Promise._settlePromises (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/promise.js:700:14)\n    at Async._drainQueue (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/async.js:123:16)\n    at Async._drainQueues (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/async.js:133:10)\n    at Immediate.Async.drainQueues (/Applications/Nylas N1.app/Contents/Resources/app.asar/node_modules/bluebird/js/main/async.js:15:14)\n    at tryOnImmediate (timers.js:554:5)\n    at processImmediate [as _immediateCallback] (timers.js:533:5)"}},"pluginIds":["message-list"]}}
```

Thanks.

**Comment by bengotow** [comment]
Wow that's a great find - thanks. It looks like one of these emails contains an inline attachment with a non Latin1 name? I'll see if I can reproduce.

**Comment by bengotow** [comment]
Reproduced and fixed in github.com/nylas/N1/commit/d95756ba98e899e417abf8a9c3d88cbf62a06bff, thanks!

**Comment by detouched** [comment]
Thanks for the prompt fix! :)

> It looks like one of these emails contains an inline attachment with a non Latin1 name?

Yeah, it had em dash which apparently is not in ISO-8859-1.

**Comment by emorikawa** [comment]
Fixed with 0.4.400 release
